### PR TITLE
Ingawei/move group search to supabase

### PR DESCRIPTION
**Pull request by ingawei**

### CODE REVIEW

File: web/lib/supabase/contract-groups.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+      run(
+        db
+          .from('groups')
+          .select('*')
+          .or('data->>privacyStatus.eq.public,data->>privacyStatus.eq.curated')
+          .ilike('data->>name', `%${prompt}%`)
+          .order('data->totalMembers', { ascending: false } as any)
+          .limit(GROUPS_LIST_SIZE)
+      ),
+    ])
```
Comment (jahooma) [inline web/lib/supabase/contract-groups.ts:99]: I think a helper function could make this easier.

You could build most of the query once except for the `.ilike()` and then use a helper function to do the parallel queries, e.g.:

```
function (q, field, prompt) {
  return await Promise.all([
  q.ilike(field, prompt),
  q.ilike(field, `{prompt}%`),
  q.ilike(field, `%{prompt}`%),
}
```

File: web/components/activity-log.tsx [hunk 2]
```diff
@@ -37,14 +37,17 @@ export function ActivityLog(props: {
...
-    )
+  useEffect(() => {
+    if (user) {
+      getShouldBlockDestiny(user.id).then((result) =>
+        setShouldBlockDestiny(result)
+      )
+    } else if (!shouldBlockDestiny) {
+      setShouldBlockDestiny(true)
+    }
+  }, [user?.id])
```
Comment (jahooma) [inline web/components/activity-log.tsx:50]: You could make this a hook `useShouldBlockDestinty`. And then use it in `use-feed.ts` as well. 
This wouldn't block this PR though.

File: common/src/supabase.schema.ts [hunk 3]
```diff
@@ -0,0 +1,1113 @@
+export type Json =
```
Comment (jahooma) [inline common/src/supabase.schema.ts:1]: I don't think we need this copy of the schema here, right? You can delete this file.

**Review by jahooma — APPROVED** [review]
Looks good, Inga! Just a couple comments.

File: web/lib/supabase/contract-groups.ts (hunk 1 above)
Comment (ingawei) [reply web/lib/supabase/contract-groups.ts:99]: could not do :(

**Pull request merged**